aspnetcore/migration/mvc.md

```diff
@@ -3,18 +3,189 @@ title: Migrate from ASP.NET MVC to ASP.NET Core MVC
+This article shows how to get started migrating an ASP.NET MVC project to [ASP.NET Core MVC](../mvc/overview.md). In the process, it highlights related changes from ASP.NET MVC.
+
+Migrating from ASP.NET MVC is a multiple step process and this article covers:
+
+* The initial setup.
+* Basic controllers and views.
+* Static content.
+* Client-side dependencies.
+
+For migrating configuration and identity code, see [Migrate configuration to ASP.NET Core](configuration.md) and [Migrate Authentication and Identity to ASP.NET Core](identity.md).
+
+## Prerequisites
+
+[!INCLUDE [prerequisites](../includes/net-core-prereqs-vs-3.1.md)]
+
+## Create the starter ASP.NET MVC project
+
+Create an example ASP.NET MVC project in Visual Studio to migrate:
+
+1. From the **File** menu, select **New** > **Project**.
+1. Select **ASP.NET Web Application (.NET Framework)** and then select **Next**.
+1. Name the project *WebApp1* so the namespace matches the ASP.NET Core project created in the next step. Select **Create**.
+1. Select **MVC**, and then select **Create**.
+
+## Create the ASP.NET Core project
+
+Create a new solution with a new ASP.NET Core project to migrate to:
+
+1. Launch a second instance of Visual Studio.
+1. From the **File** menu, select **New** > **Project**.
+1. Select **ASP.NET Web Core Web Application** and then select **Next**.
+1. In the **Configure your new project** dialog, Name the project *WebApp1*.
+1. Set the location to a different directory than the previous project to use the same project name. Having the same namespace makes it easier to copy code between the two projects. Select **Create**.
+1. In the **Create a new ASP.NET Core Web Application** dialog, confirm that **.NET Core** and **ASP.NET Core 3.1** are selected. Select the **Web Application (Model-View-Controller)** project template, and select **Create**.
+
+## Configure the ASP.NET Core site to use MVC
+
+When targeting .NET Core, the [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapackage-app) is referenced by default. This package contains packages commonly used by MVC apps. If targeting .NET Framework, package references must be listed individually in the project file. `Microsoft.AspNetCore.Mvc` is the ASP.NET Core MVC framework.
+
+In ASP.NET Core, the `Startup` class:
+
+* Replaces *Global.asax*.
+* Handles all app startup tasks.
+
+For more information, see <xref:fundamentals/startup>.
+
+* In the ASP.NET Core project, open the *Startup.cs* file.
+
+  [!code-csharp[](mvc/samples/3.x/Startup.cs?highlight=26,43,45)]
+
+The ASP.NET Core runtime is modular, and ASP.NET Core 3.0 added new options for registering MVC scenarios inside *Startup.ConfigureServices*.  You must also explicitly opt in for features such as serving static files and routing.
+
+In the previous code, the **Web Application (Model-View-Controller)** project template opts in the following MVC Scenario and features:
```

Comment:
I agree it would make it slimmer and the information can be found in the startup doc, however, I think it is worth keeping this section of information here.  It is a typical rather than fringe step in migration to determine what you need to opt in for your ASP.NET MVC project based on what your previous project was using, since they need to opt in now.  I think we hit the "80% rule" in this case and shouldn't have them have to switch documents and come back.

Reply:
This write up won't tell you what you need to migrate, a link to startup helps, but even that is not enough. It's not a scenario and feature, it's a service and middleware. The doc is migration, and the brief description doesn't really help migration.

Reply:
I question the value of most of this doc. It's really, how to migrate the simplest possible ASP.NET app to the simplest possible ASP.NET Core app. A real app will be so much more complicated,  good links to the details are what's important.

Not only are we converting the simplest possible app, we're not even including auth.

Reply:
I was referring to removing the section, which I think works better to keep and I found it helpful.  The link you mentioned is helpful too.  I'll add that.